packages/components/src/tooltip/index.js

```diff
@@ -24,60 +25,69 @@ import Shortcut from '../shortcut';
  *
  * @type {number}
  */
-const TOOLTIP_DELAY = 700;
-
-class Tooltip extends Component {
-	constructor() {
-		super( ...arguments );
-
-		this.delayedSetIsOver = debounce(
-			( isOver ) => this.setState( { isOver } ),
-			TOOLTIP_DELAY
-		);
-
-		/**
-		 * Prebound `isInMouseDown` handler, created as a constant reference to
-		 * assure ability to remove in component unmount.
-		 *
-		 * @type {Function}
-		 */
-		this.cancelIsMouseDown = this.createSetIsMouseDown( false );
-
-		/**
-		 * Whether a the mouse is currently pressed, used in determining whether
-		 * to handle a focus event as displaying the tooltip immediately.
-		 *
-		 * @type {boolean}
-		 */
-		this.isInMouseDown = false;
-
-		this.state = {
-			isOver: false,
-		};
-	}
+export const TOOLTIP_DELAY = 700;
 
-	componentWillUnmount() {
-		this.delayedSetIsOver.cancel();
+const emitToChild = ( children, eventName, event ) => {
+	if ( Children.count( children ) !== 1 ) {
+		return;
+	}
 
-		document.removeEventListener( 'mouseup', this.cancelIsMouseDown );
+	const child = Children.only( children );
+	if ( typeof child.props[ eventName ] === 'function' ) {
+		child.props[ eventName ]( event );
 	}
+};
 
-	emitToChild( eventName, event ) {
-		const { children } = this.props;
-		if ( Children.count( children ) !== 1 ) {
-			return;
-		}
+function Tooltip( { children, position, text, shortcut } ) {
+	/**
+	 * Whether a mouse is currently pressed, used in determining whether
+	 * to handle a focus event as displaying the tooltip immediately.
+	 *
+	 * @type {boolean}
+	 */
+	const [ isMouseDown, setIsMouseDown ] = useState( false );
+	const [ isOver, setIsOver ] = useState( false );
+	const delayedSetIsOver = debounce( setIsOver, TOOLTIP_DELAY );
 
-		const child = Children.only( children );
-		if ( typeof child.props[ eventName ] === 'function' ) {
-			child.props[ eventName ]( event );
-		}
-	}
+	/**
+	 * Creates an event callback to handle assignment of the `isInMouseDown`
+	 * instance property in response to a `mousedown` or `mouseup` event.
+	 *
+	 * @param {boolean} isForMouseDown Whether handler is to be created for the
+	 *                              `mousedown` event, as opposed to `mouseup`.
+	 *
+	 * @return {Function} Event callback handler.
+	 */
+	const createSetIsMouseDown = ( isForMouseDown ) => {
+		const eventType = isForMouseDown ? 'onMouseDown' : 'onMouseUp';
+		const eventListenerAction =
+			( isForMouseDown ? 'add' : 'remove' ) + 'EventListener';
+		return ( event ) => {
+			// Preserve original child callback behavior
+			emitToChild( children, eventType, event );
+
+			// On mouse down, the next `mouseup` should revert the value of the
+			// instance property and remove its own event handler. The bind is
+			// made on the document since the `mouseup` might not occur within
+			// the bounds of the element.
+			document[ eventListenerAction ]( 'mouseup', cancelIsMouseDown );
+
+			setIsMouseDown( isForMouseDown );
+		};
+	};
```

Comment:
This is more of an issue with the previous code than your code, but I find the way this code was written in a non-repetitive (DRY) way has made it quite hard to read (and isn't actually much less code).

I wonder if it'd be simpler if they were separate functions for `onMouseDown` and `onMouseUp`. What do you think?